src/write.c

```diff
@@ -415,6 +425,10 @@ avifEncoder * avifEncoderCreate(void)
     encoder->scalingMode = noScaling;
     encoder->data = avifEncoderDataCreate();
     encoder->csOptions = avifCodecSpecificOptionsCreate();
+    if (!encoder->data || !encoder->csOptions) {
+        avifEncoderDestroy(encoder);
+        return NULL;
+    }
```

Comment:
`avifEncoderDestroy()` needs to handle a null `encoder->csOptions` or `encoder->data` now:

```
void avifEncoderDestroy(avifEncoder * encoder)
{
    // NOTE: avifCodecSpecificOptionsDestroy() already allows its input to be null,
    // so this if check isn't really necessary. I probably would remove the null pointer
    // check from avifCodecSpecificOptionsDestroy() and check it here.
    if (encoder->csOptions) {
        avifCodecSpecificOptionsDestroy(encoder->csOptions);
    }
    // NOTE: It is tempting to change avifEncoderDataDestroy() to allow a null point.
    // I prefer an explicit null check at call site so that we only allow a null pointer
    // where it is expected. But this doesn't really matter.
    if (encoder->data) {
        avifEncoderDataDestroy(encoder->data);
    }
    avifFree(encoder);
}
```

NOTE: git history showed that the null pointer check in `avifCodecSpecificOptionsDestroy()` was there when Joe first added this function in https://github.com/AOMediaCodec/libavif/commit/e1097bd2. I don't know why he added the null pointer check, because it wasn't necessary and also it wasn't present in most of the destroy functions in libavif.